### PR TITLE
chore(backend-feedback): Incorporate LFA back-end feedback

### PR DESCRIPTION
**Pull request by Bastieno**
#### What does this PR do?
* Incorporate LFA backend feedback

 #### Description of Task to be completed?
* Modify signin route to allow the use of token for signing in
* Implement an endpoint to change the '_isAdmin_' property of a user (i.e. to make a user an admin)
* Modify the base url to make `/api/v1` a valid route
* Write tests to cover these implementations
* Use of _sinon_ and _sinon-chai_ in writing some of my tests. I was able to increase the test coverage. Check the screenshot below.
* Write swagger docs for app
* Make some miscellaneous fixes where necessary

#### How should this be manually tested?
* Clone the repo `git clone https://github.com/Bastieno/QuickCredit.git`
* Checkout to the branch `git checkout ch-lfa-backend-feedback-166173934`
* Install Postgresql
* Create a database named quickcreditdb_test
* Run `npm install` to install application dependencies
* Test by `npm test`

#### Screenshots (if appropriate)
test_result1

#### Questions:
N/A

#### Any background context you want to provide?
I tried as much to do all the feedback to the letter. I had some problem using _sinon_ because some of the tests I wrote using _sinon_ was passing but I could really see the effect in terms of test coverage.

#### What are the relevant pivotal tracker stories?
[#166173934](https://www.pivotaltracker.com/n/projects/2328087)

### CODE REVIEW

File: server/app.js [hunk 1]
```diff
@@ -19,7 +19,7 @@ app.use(router);
 
 // Implement the 'catch-all' errorHandler
 app.use((err, req, res, next) => {
-  res.status(err.status || 500);
+  res.status(500);
   res.json({
     message: `Hey!! we caught the error 👍👍, ${err.stack} `,
```
Comment (Oluwasegun-AA) [inline server/app.js]: Nice Job @Bastieno , Please make your server error message more professional

File: package.json [hunk 2]
```diff
@@ -69,6 +69,8 @@
     "eslint-config-airbnb-base": "^13.1.0",
     "eslint-plugin-import": "^2.16.0",
     "nodemon": "^1.18.11",
-    "nyc": "^13.3.0"
+    "nyc": "^13.3.0",
+    "sinon": "^7.3.2",
+    "sinon-chai": "^3.3.0"
```
Comment (Oluwasegun-AA) [inline package.json]: Please lock all your file versions by removing the caret symbols

File: server/routes/user/index.js [hunk 3]
```diff
@@ -2,6 +2,13 @@ import { Router } from 'express';
 import { auth, users } from './user';
 
 const router = Router();
+
+router.get('/', (req, res) => {
```
Comment (Oluwasegun-AA) [inline server/routes/user/index.js:6]: you should only get welcome to quick credit on `api/v1`

File: server/test/02-utils/responseHandler.test.js [hunk 4]
```diff
@@ -1,9 +1,21 @@
...
 describe('Response Handler Utility Function', () => {
   it('The utility file should export a function', (done) => {
     expect(responseHandler).to.be.a('function');
     done();
   });
+
+  it('It should return an error', (done) => {
+    const result = new Error();
+    const res = {};
+    const next = sinon.spy();
```
Comment (Oluwasegun-AA) [inline server/test/02-utils/responseHandler.test.js:14]: Nice trial with Sinon

File: server/test/04-misc/misc.test.js [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+  // ADD CODE HERE
+  it('fakes server error', async () => {
+    app.use((err = new Error(), req = {}, res = { status() {}, json() {} }, next = sinon.spy()) => {
+      sinon.stub(res, 'status').returnsThis();
+      expect(res.status).to.have.been.calledWith(500);
+      expect(next).to.have.not.been.calledOnce;
+    });
+  });
+*/
+});
```
Comment (Oluwasegun-AA) [inline server/test/04-misc/misc.test.js:37]: Please remove every commented lines of code in your PR

File: package.json (hunk 2 above)
Comment (Oluwasegun-AA) [reply package.json]: caret symbol === ˆ

File: server/middleware/auth.js [hunk 6]
```diff
@@ -38,13 +87,15 @@ const adminOnly = (req, res, next) => {
 
 const userOnly = (req, res, next) => {
   const { userId } = req.user;
-  if (!userId) {
-    return res.status(403).send({
-      status: 403,
-      error: 'Unauthorized Access. For authenticated users only',
-    });
-  }
+  // Fix this: doing nothing!
```
Comment (Oluwasegun-AA) [inline server/middleware/auth.js]: @Bastieno can you please attend to this redundancy in the code base?

File: server/app.js (hunk 1 above)
Comment (Bastieno) [reply server/app.js]: I have changed the server error message

File: server/routes/user/index.js (hunk 3 above)
Comment (Bastieno) [reply server/routes/user/index.js:6]: You suggested I keep the router.get('/') and router.get('/api/v1') to have the same message:
_Welcome to QuickCredit_

File: package.json (hunk 2 above)
Comment (Bastieno) [reply package.json]: I removed the caret symbol from my package.json. I tried to push after doing that and it caused my build to fail.
I had to do `npm install` to reinstall all my dependencies before my build passed.

File: server/test/04-misc/misc.test.js (hunk 5 above)
Comment (Bastieno) [reply server/test/04-misc/misc.test.js:37]: I have removed all commented lines of codes.

File: server/middleware/auth.js (hunk 6 above)
Comment (Bastieno) [reply server/middleware/auth.js]: Done that.

**Pull request merged**